test/startservers.py

```diff
@@ -10,6 +11,132 @@
 
 from helpers import waitport, config_dir, CONFIG_NEXT
 
+Service = collections.namedtuple('Service', ('name', 'port', 'cmd', 'deps'))
+
+SERVICES = (
+    Service('bad-key-revoker',
+        8020,
+        ('./bin/bad-key-revoker', '--config', os.path.join(config_dir, 'bad-key-revoker.json')),
+        None),
+    Service('boulder-remoteva-a',
+        8011,
+        ('./bin/boulder-remoteva', '--config', os.path.join(config_dir, 'va-remote-a.json')),
+        ('bad-key-revoker',)),
+    Service('boulder-remoteva-b',
+        8012,
+        ('./bin/boulder-remoteva', '--config', os.path.join(config_dir, 'va-remote-b.json')),
+        ('bad-key-revoker',)),
+    Service('sd-test-srv',
+        8020,
+        ('./bin/sd-test-srv', '--listen', ':53'),
+        ('boulder-remoteva-a', 'boulder-remoteva-b')),
+    Service('boulder-sa-1',
+        8003,
+        ('./bin/boulder-sa', '--config', os.path.join(config_dir, 'sa.json'), '--addr', 'sa1.boulder:9095', '--debug-addr', ':8003'),
+        ('sd-test-srv',)),
+    Service('boulder-sa-2',
+        8103,
+        ('./bin/boulder-sa', '--config', os.path.join(config_dir, 'sa.json'), '--addr', 'sa2.boulder:9095', '--debug-addr', ':8103'),
+        ('sd-test-srv',)),
+    Service('ct-test-srv',
+        4500,
+        ('./bin/ct-test-srv', '--config', 'test/ct-test-srv/ct-test-srv.json'),
+        ('boulder-sa-1', 'boulder-sa-2')),
+    Service('boulder-publisher-1',
+        8009,
+        ('./bin/boulder-publisher', '--config', os.path.join(config_dir, 'publisher.json'), '--addr', 'publisher1.boulder:9091', '--debug-addr', ':8009'),
+        ('ct-test-srv',)),
+    Service('boulder-publisher-2',
+        8109,
+        ('./bin/boulder-publisher', '--config', os.path.join(config_dir, 'publisher.json'), '--addr', 'publisher2.boulder:9091', '--debug-addr', ':8109'),
+        ('ct-test-srv',)),
+    Service('mail-test-srv',
+        9380,
+        ('./bin/mail-test-srv', '--closeFirst', '5', '--cert', 'test/mail-test-srv/localhost/cert.pem', '--key', 'test/mail-test-srv/localhost/key.pem'),
+        ('boulder-publisher-1', 'boulder-publisher-2')),
+    Service('ocsp-responder',
+        8005,
+        ('./bin/ocsp-responder', '--config', os.path.join(config_dir, 'ocsp-responder.json')),
+        ('mail-test-srv',)),
+    Service('boulder-va-1',
+        8004,
+        ('./bin/boulder-va', '--config', os.path.join(config_dir, 'va.json'), '--addr', 'va1.boulder:9092', '--debug-addr', ':8004'),
+        ('ocsp-responder',)),
+    Service('boulder-va-2',
+        8104,
+        ('./bin/boulder-va', '--config', os.path.join(config_dir, 'va.json'), '--addr', 'va2.boulder:9092', '--debug-addr', ':8104'),
+        ('ocsp-responder',)),
+    Service('boulder-ca-a',
+        8004,
+        ('./bin/boulder-ca', '--config', os.path.join(config_dir, 'ca-a.json'), '--ca-addr', 'ca1.boulder:9093', '--ocsp-addr', 'ca1.boulder:9096', '--debug-addr', ':8001'),
+        ('boulder-va-1', 'boulder-va-2')),
```

Comment:
The CA doesn't depend on the VA, it depends on the SA. Also, I think these names don't exist in the service definition; does the toposort need to error out on unrecognized dependencies?

Reply:
I was incorrect about boulder-va-1 not existing. It looks like the toposort is doing the correct thing.